### ISSUE TITLE
Specification of binary data format

### ISSUE THREAD

**Issue by ept** [automerge/automerge#297]
We need a detailed documentation/specification of Automerge's new binary data format (#253), so that others can implement it if they want to. I am thinking an RFC-style document of prose, structure defintions, and pseudocode. One option would be to use the [presentation language from RFC8446](https://tools.ietf.org/html/rfc8446#section-3).

**Comment by alexjg** [comment]
I think if the goal is for alternative implementations this should probably be broader than just the binary format, other implementations will also need to know how to merge opertions no?

**Comment by ept** [comment]
Yes! And it will also need to explain how to convert between the individual changes and the whole-document compression, in both directions.
